src/main/scala/io/iohk/ethereum/consensus/ConsensusAdapter.scala

```diff
@@ -35,15 +50,54 @@ class ConsensusAdapter(
         if (isBlockADuplicate(block.header, bestBlock.header.number)) {
           log.debug("Ignoring duplicated block: {}", block.idTag)
           Task.now(DuplicateBlock)
+        } else if (blockchain.getChainWeightByHash(bestBlock.header.hash).isEmpty) {
+          // This part is not really needed except for compatibility as a missing chain weight
+          // would indicate an inconsistent database
+          returnNoTotalDifficulty(bestBlock)
         } else {
           doBlockPreValidation(block).flatMap {
             case Left(error) =>
               Task.now(BlockImportFailed(error.reason.toString))
             case Right(BlockExecutionSuccess) =>
-              consensus.evaluateBranch(Seq(block))
+              enqueueAndGetBranch(block, bestBlock.number) match {
+                case None =>
+                  Task.now(BlockEnqueued)
+                case Some(newBranch) =>
+                  consensus
+                    .evaluateBranch(newBranch)
```

Comment:
This method got really big, can you maybe separate this whole chunk of branch evaluation?

Reply:
I extracted the big match in another method.